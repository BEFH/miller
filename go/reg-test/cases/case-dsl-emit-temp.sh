announce "1. NON-LASHED NON-INDEXED NAMEDVAR"

run_mlr --ojson --jvstack --from $indir/abixy put -q '
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp @sumx;
    emit  @sumx;
  }
'

announce "1. NON-LASHED NON-INDEXED MAP"

run_mlr --ojson --jvstack --from $indir/abixy put -q '
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp @sums;
    emit  @sums;
  }
'

announce "2. LASHED NON-INDEXED NAMEDVAR"
run_mlr --ojson --jvstack --from $indir/abixy put -q '
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp (@sumx, @countx);
    emit  (@sumx, @countx);
  }
'

announce "2. LASHED NON-INDEXED MAP"
run_mlr --ojson --jvstack --from $indir/abixy put -q '
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp (@sums, @counts);
    emit  (@sums, @counts);
  }
'

announce "3. NON-LASHED INDEXED MAP"
run_mlr --ojson --jvstack --from $indir/abixy put -q '
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp @sums, "a";
    emit  @sums, "a";
  }
'

announce "4. LASHED INDEXED MAP"
run_mlr --ojson --jvstack --from $indir/abixy put -q '
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp (@sums, @counts), "a";
    emit  (@sums, @counts), "a";
  }
'




