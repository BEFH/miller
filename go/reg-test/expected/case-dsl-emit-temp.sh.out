

================================================================
1. NON-LASHED NON-INDEXED NAMEDVAR

mlr --ojson --jvstack --from ./reg-test/input/abixy put -q 
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp @sumx;
    emit  @sumx;
  }

{
  "sumx": 4.536293840335763
}
{
  "sumx": 4.536293840335763
}


================================================================
1. NON-LASHED NON-INDEXED MAP

mlr --ojson --jvstack --from ./reg-test/input/abixy put -q 
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp @sums;
    emit  @sums;
  }

{
  "sums": {
    "pan": 0.8494161498792961,
    "eks": 1.75186341922895,
    "wye": 0.7778922255683036,
    "zee": 1.1256801691982772,
    "hat": 0.03144187646093577
  }
}
{
  "pan": 0.8494161498792961,
  "eks": 1.75186341922895,
  "wye": 0.7778922255683036,
  "zee": 1.1256801691982772,
  "hat": 0.03144187646093577
}


================================================================
2. LASHED NON-INDEXED NAMEDVAR

mlr --ojson --jvstack --from ./reg-test/input/abixy put -q 
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp (@sumx, @countx);
    emit  (@sumx, @countx);
  }

{
  "sumx": 4.536293840335763,
  "countx": 10
}
{
  "sumx": 4.536293840335763,
  "countx": 10
}


================================================================
2. LASHED NON-INDEXED MAP

mlr --ojson --jvstack --from ./reg-test/input/abixy put -q 
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp (@sums, @counts);
    emit  (@sums, @counts);
  }

{
  "sums": {
    "pan": 0.8494161498792961,
    "eks": 1.75186341922895,
    "wye": 0.7778922255683036,
    "zee": 1.1256801691982772,
    "hat": 0.03144187646093577
  },
  "counts": {
    "pan": 2,
    "eks": 3,
    "wye": 2,
    "zee": 2,
    "hat": 1
  }
}
{
  "pan": 2,
  "eks": 3,
  "wye": 2,
  "zee": 2,
  "hat": 1
}


================================================================
3. NON-LASHED INDEXED MAP

mlr --ojson --jvstack --from ./reg-test/input/abixy put -q 
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp @sums, "a";
    emit  @sums, "a";
  }

{
  "a": "pan",
  "sums": 0.8494161498792961
}
{
  "a": "eks",
  "sums": 1.75186341922895
}
{
  "a": "wye",
  "sums": 0.7778922255683036
}
{
  "a": "zee",
  "sums": 1.1256801691982772
}
{
  "a": "hat",
  "sums": 0.03144187646093577
}
{
  "a": "pan",
  "sums": 0.8494161498792961
}
{
  "a": "eks",
  "sums": 1.75186341922895
}
{
  "a": "wye",
  "sums": 0.7778922255683036
}
{
  "a": "zee",
  "sums": 1.1256801691982772
}
{
  "a": "hat",
  "sums": 0.03144187646093577
}


================================================================
4. LASHED INDEXED MAP

mlr --ojson --jvstack --from ./reg-test/input/abixy put -q 
  @sumx += $x;
  @countx += 1;
  @sums[$a] += $x;
  @counts[$a] += 1;
  end {
    emitp (@sums, @counts), "a";
    emit  (@sums, @counts), "a";
  }

{
  "a": "pan",
  "sums": 0.8494161498792961,
  "counts": 2
}
{
  "a": "eks",
  "sums": 1.75186341922895,
  "counts": 3
}
{
  "a": "wye",
  "sums": 0.7778922255683036,
  "counts": 2
}
{
  "a": "zee",
  "sums": 1.1256801691982772,
  "counts": 2
}
{
  "a": "hat",
  "sums": 0.03144187646093577,
  "counts": 1
}
{
  "a": "pan",
  "sums": 0.8494161498792961,
  "counts": 2
}
{
  "a": "eks",
  "sums": 1.75186341922895,
  "counts": 3
}
{
  "a": "wye",
  "sums": 0.7778922255683036,
  "counts": 2
}
{
  "a": "zee",
  "sums": 1.1256801691982772,
  "counts": 2
}
{
  "a": "hat",
  "sums": 0.03144187646093577,
  "counts": 1
}

